File: src/main/java/org/apache/prepbuddy/datacleansers/imputation/ImputationStrategy.java
package org.apache.prepbuddy.datacleansers.imputation;

import org.apache.prepbuddy.rdds.TransformableRDD;
import org.apache.prepbuddy.utils.RowRecord;

import java.io.Serializable;

public interface ImputationStrategy extends Serializable {
    void prepareSubstitute(TransformableRDD rdd, int missingDataColumn);

    String handleMissingData(RowRecord record);
}

File: src/main/scala/org/apache/datacommons/prepbuddy/imputation/ImputationStrategy.scala
package org.apache.datacommons.prepbuddy.imputation

import org.apache.datacommons.prepbuddy.rdds.TransformableRDD
import org.apache.datacommons.prepbuddy.utils.RowRecord

trait ImputationStrategy extends Serializable{
  def prepareSubstitute(rdd: TransformableRDD, missingDataColumn: Int)

  def handleMissingData(record: RowRecord): String
}
